src/main/java/seedu/address/model/module/DistinctModuleList.java

```diff
@@ -52,6 +53,46 @@ public void setModules(List<Module> modules) {
         moduleList.setAll(modules);
     }
 
+    /**
+     * Counts the number of tasks in {@code tasks} that belong to {@code module},
+     * and updates this number in {@code module}.
+     * {@code module} must exist in the module list.
+     *
+     * @param module The module to check for number of tasks.
+     * @param tasks the list of tasks to check with the module.
+     */
+    public void updateTotalNumOfTasks(Module module, DistinctTaskList tasks) {
+        int totalNumOfTasks = tasks.getTotalNumOfTasks(module);
```

Comment:
Perhaps you could look into including null checks for both module and DistinctTaskList here(More about defensive programming)

Reply:
I agree with you, I have included the null checks.